### DISCUSSION TITLE
feat: add custom HTTP headers to each session of ollama-webui

### DISCUSSION THREAD

**Discussion by mjtechguy** [open-webui/open-webui#4814] [General]
**Is your feature request related to a problem? Please describe.**
No problem.

**Describe the solution you'd like**
I would like to allow for a cookie to be passed to the ollama-webui session that my user is logged into. This would allow for me to pin each user session to a specific ollama server for an amount of time when running multiple ollama servers behind haproxy.

**Describe alternatives you've considered**
None of the other solutions seems feasible. They are based on IP, but all sessions of the Ollama webui come from the same IP, regardless of user.

UPDATE: See better solution for HTTP header support mentioned below.

**Comment by justinh-rahb** [comment]
Considering the specific communication flow between the Ollama WebUI and Ollama (via HAProxy), it appears that utilizing a cookie for session persistence might not be the optimal approach. Given that cookies are primarily a client-side mechanism designed for browser-server interactions, your scenario, which involves server-to-server communication, would be better served by leveraging custom HTTP headers sent from the backend. These headers can carry unique session IDs, enabling HAProxy to accurately route requests to the appropriate server based on session affinity, and should be ignored by Ollama normally if there's no proxy in play.

**Comment by mjtechguy** [comment]
I stand corrected. That does seem like a much better solution. Thanks @justinh-rahb

**Comment by justinh-rahb** [comment]
Don't mention it @mjtechguy, I understood what you intended for the outcome, just helping to make sure it's articulated properly for everyone👌

**Comment by smartynov** [comment]
Any updates on this? We faced similar problem, adding custom header to both ollama and openai api requests seems to be a good solution. Is it going to be implemented anytime soon?

**Comment by smartynov** [comment]
I stumbled upon [ENABLE_FORWARD_USER_INFO_HEADERS](https://docs.openwebui.com/getting-started/env-configuration/#enable_forward_user_info_headers) option, and it solves most of the problem. Setting ENABLE_FORWARD_USER_INFO_HEADERS=True in env makes open-webui send additional headers suitable for user identification.

**Comment by jakobhanna** [comment]
Hi, i would say that for load balancing with HAProxy a header with the chat-id would be good. For sticky sessions the chat id is way better then user infos. But i can not figure out a way to pass this header or add it in the OWUI backend.
